nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -1172,14 +1172,17 @@ def loss_func(self, loss_mask, num_valid_tokens_in_ub, output_tensor):
         return loss
 
     def build_train_valid_test_datasets(self):
-        # Override limit_val_batches to be a multiple of num microbatches to prevent val_step from exiting in between a step
-        self._reconfigure_val_batches()
-        logging.info('Building GPT datasets.')
         if self.trainer.limit_val_batches > 1.0 and isinstance(self.trainer.limit_val_batches, float):
             raise ValueError("limit_val_batches must be an integer or float less than or equal to 1.0.")
+        logging.info('Building GPT datasets.')
         global_batch_size = self.cfg.global_batch_size
         max_train_steps = self.trainer.max_steps
-        eval_iters = (max_train_steps // self.trainer.val_check_interval + 1) * self.trainer.limit_val_batches
+        # if limit_val_batches is 0, don't use it for computing eval samples, as it can cause error in building the dataset with 0 samples
+        eval_iters = (
```

Comment:
Separating arguments from program logic is a good practice. You could move all this arithmetic to the argument layer, as none of these variables has to be computed in the runtime.

Reply:
Yes agreed, in case of NeMo this can probably be moved inside the `__init__` of `MegatronBaseModel`.